3-enrich/spark-enrich/src/test/scala/com.snowplowanalytics.snowplow.enrich.spark/good/IabEnrichmentSpiderCfLineSpec.scala

```diff
@@ -0,0 +1,183 @@
+    null, // Transaction fields empty
+    null, //
+    null, //
+    null, //
+    null, //
+    null, //
+    null, //
+    null, //
+    null, // Transaction item fields empty
+    null, //
+    null, //
+    null, //
+    null, //
+    null, //
+    null, // Page ping fields are empty
+    null, //
+    null, //
+    null, //
+    "Mozilla/5.0 AppleWebKit/537.36 (KHTML, like Gecko; compatible; Googlebot/2.1; +http://www.google.com/bot.html) Safari/537.36",
+    "Robot/Spider",
+    "Robot/Spider",
+    null,
+    "Robot",
+    "OTHER",
+    "en-GB",
+    "1",
+    "1",
+    "1",
+    "0",
+    "0",
+    "0",
+    "0",
+    "0",
+    "1",
+    "1",
+    "32",
+    "2560",
+    "934",
+    "Unknown",
+    "Unknown",
+    "Other",
+    "Europe/London",
+    "Unknown",
+    "0",
+    "2560",
+    "1440",
+    "UTF-8",
+    "2543",
+    "1420",
+    null,
+    null,
+    null,
+    null,
+    null,
+    null,
+    null,
+    "America/Los_Angeles",
+    null,
+    null,
+    null,
+    null,
+    null,
+    null,
+    """{"schema":"iglu:com.snowplowanalytics.snowplow/contexts/jsonschema/1-0-1","data":[{"schema":"iglu:com.snowplowanalytics.snowplow/ua_parser_context/jsonschema/1-0-0","data":{"useragentFamily":"Googlebot","useragentMajor":"2","useragentMinor":"1","useragentPatch":null,"useragentVersion":"Googlebot 2.1","osFamily":"Other","osMajor":null,"osMinor":null,"osPatch":null,"osPatchMinor":null,"osVersion":"Other","deviceFamily":"Spider"}},{"schema":"iglu:com.iab.snowplow/iab_spiders_and_robots/jsonschema/1-0-0","data":{"spiderOrRobot":true,"category":"SPIDER_OR_ROBOT","reason":"FAILED_UA_INCLUDE","primaryImpact":"UNKNOWN"}}]}""",
+    null,
+    "2012-05-27 11:35:53.000",
+    "com.snowplowanalytics.snowplow.input-adapters",
+    "segment_webhook_config",
+    "jsonschema",
+    "1-0-0",
+    "6a2f7bbf7669eb1808c79cfe3e1f8e57"
+  )
+}
+
+/**
+ * Check that the IAB enrichment successfully atttaches a context
+ * to an event.
+ */
+class IabEnrichmentSpiderCfLineSpec extends Specification with EnrichJobSpec {
+  import EnrichJobSpec._
+  override def appName = "iab-enrichment-cf-lines"
+  sequential
+  "A job which processes a CloudFront file containing 1 valid custom unstructured event" should {
+    runEnrichJob(IabEnrichmentSpiderCfLineSpec.lines, "cloudfront", "1", false, List("geo", "domain"), iabEnrichmentEnabled = true)
+
+    "correctly attach a derived context fetched from the IAB enrichment" in {
+      val Some(goods) = readPartFile(dirs.output)
+      goods.size must_== 1
+      val actual = goods.head.split("\t").map(s => if (s.isEmpty()) null else s)
+      for (idx <- IabEnrichmentSpiderCfLineSpec.expected.indices) {
+        actual(idx) must BeFieldEqualTo(IabEnrichmentSpiderCfLineSpec.expected(idx), idx)
+      }
+    }
+
+    "not write any bad rows" in {
+      dirs.badRows must beEmptyDir
+    }
+  }
```

Comment:
can we have the two iab specs turned into one spec? No need to have two different files for the same enrichment.